cnd/src/http_api/routes/rfc003/handlers/post_swap.rs

```diff
@@ -20,35 +20,35 @@ use crate::{
 use anyhow::Context;
 use futures_core::future::TryFutureExt;
 use libp2p_comit::frame::OutboundRequest;
-use serde::{Deserialize, Serialize};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use std::{convert::TryInto, fmt::Debug, str::FromStr};
 
-async fn initiate_request<AL, BL, AA, BA>(
+async fn initiate_request<'de, AL, BL, AA, BA, AI, BI>(
     dependencies: Facade,
     id: SwapId,
     peer: DialInformation,
-    swap_request: rfc003::Request<AL, BL, AA, BA>,
+    swap_request: rfc003::Request<AL, BL, AA, BA, AI, BI>,
 ) -> anyhow::Result<()>
 where
-    Sqlite: Save<Request<AL, BL, AA, BA>>
-        + Save<Accept<AL::Identity, BL::Identity>>
-        + Save<Swap>
-        + Save<Decline>,
+    Sqlite:
+        Save<Request<AL, BL, AA, BA, AI, BI>> + Save<Accept<AI, BI>> + Save<Swap> + Save<Decline>,
     AL: Ledger,
     BL: Ledger,
     AA: Ord + Send + Sync + 'static,
     BA: Ord + Send + Sync + 'static,
-    rfc003::Request<AL, BL, AA, BA>: TryInto<OutboundRequest> + Clone,
-    <rfc003::Request<AL, BL, AA, BA> as TryInto<OutboundRequest>>::Error: Debug,
-    Facade: LoadAcceptedSwap<AL, BL, AA, BA>
-        + HtlcFunded<AL, AA>
-        + HtlcFunded<BL, BA>
-        + HtlcDeployed<AL, AA>
-        + HtlcDeployed<BL, BA>
-        + HtlcRedeemed<AL, AA>
-        + HtlcRedeemed<BL, BA>
-        + HtlcRefunded<AL, AA>
-        + HtlcRefunded<BL, BA>,
+    AI: Send + Sync + 'static + DeserializeOwned + Clone,
```

Comment:
I would be better to apply the "say what you actually mean"-rule here. I think we just need `Accept<AI, BI>: DeserializeOwned`.

Reply:
Also, please try to remove the `Clone` bound and specify the type you are actually calling `Clone` on.